src/lib.rs

```diff
@@ -243,3 +246,9 @@ pub use crate::term_like::TermLike;
 
 #[cfg(feature = "rayon")]
 pub use crate::rayon::ParallelProgressIterator;
+
+pub trait StatefulFormatter: Display + Send {
+    fn default(&self) -> Box<dyn StatefulFormatter>;
+    fn tick(&mut self, state: &mut ProgressState, now: Instant);
```

Comment:
I think `tick()` and `reset()` should take `&ProgressState`, not `&mut`.

Reply:
done